Modules/Product/Http/Services/ProductService.php

```diff
@@ -1178,6 +1333,39 @@ private function variables($request): array
         );
     }
 
+    private function curlCall($method, $api_key, $api_password, $store_url, $extendUrl=""): array
```

Comment:
For all the external API calls, please make separate service class and then use that where ever you need, this is Product service and it should contains only product related data.
Make a new class and shift the code there.